qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -314,8 +314,17 @@ def _score_heuristic(self, heuristic, front_layer, extended_set, layout, swap_qu
         the remaining virtual gates that must be applied.
         """
         if heuristic == 'basic':
-            return sum(self.coupling_map.distance(*[layout[q] for q in node.qargs])
-                       for node in front_layer)
+            if len(front_layer) > 1:
+                if self.coupling_map._dist_matrix is None:
+                    self.coupling_map._compute_distance_matrix()
+                return self.coupling_map._dist_matrix[
```

Comment:
It might be better to put these lines to `dist_matrix` property in CouplingMap class (if we don't mind one function call).

Reply:
Yeah, that's a good idea. I made this change in: https://github.com/Qiskit/qiskit-terra/pull/5403/commits/8df570f740ac3377f729de0f45d2db41fd82a2ca